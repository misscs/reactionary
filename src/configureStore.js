import { applyMiddleware, createStore, compose } from 'redux'
import { composeWithDevTools } from 'redux-devtools-extension/logOnlyInProduction'
import createSagaMiddleware from 'redux-saga'

import rootReducer from 'routes/Root/rootReducer'
import rootSaga from 'routes/Root/rootSaga'

const sagaMiddleware = createSagaMiddleware()

const middlewares = [
  sagaMiddleware
]

const enhancers = [
  composeWithDevTools(applyMiddleware(...middlewares))
]

const composedEnhancer = compose(...enhancers)

/**
 * Create the store with middleware
 * @param { object } initialState - initial state
 * @return { object } store - store
 * @todo Check environment and only execute debugEnhancer if in development
 */
export default function configureStore (initialState = {}) {
  const store = createStore(
    rootReducer,
    initialState,
    composedEnhancer
  )

  // Kick off the root saga
  sagaMiddleware.run(rootSaga)

  return store
}
