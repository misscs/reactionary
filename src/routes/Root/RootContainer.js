import React from 'react'
import { connect } from 'react-redux'
import { createStructuredSelector } from 'reselect'

import Header from 'components/Header'

import {
  getLoggedIn,
  getIsFetching,
  getUsername
} from 'routes/Login/LoginSelectors'

import './Root.css'

export class RootContainer extends React.Component {
  render () {
    return (
      <div>
        <Header
          loggedIn={this.props.loggedIn}
          isFetching={this.props.isFetching}
          name={this.props.name}
        />
        <main id='container'>
          {this.props.children}
        </main>
      </div>
    )
  }
}

RootContainer.propTypes = {
  children: React.PropTypes.node
}

const mapStateToProps = createStructuredSelector({
  loggedIn: getLoggedIn(),
  isFetching: getIsFetching(),
  name: getUsername()
})

export default connect(mapStateToProps)(RootContainer)
