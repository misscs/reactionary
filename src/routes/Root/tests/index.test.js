import React from 'react'
import { Provider } from 'react-redux'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import RootContainer from 'routes/Root/RootContainer'
import configureStore from 'configureStore'

describe('<RootContainer />', () => {
  let store
  beforeEach(() => {
    store = configureStore()
  })

  it(`should render`, () => {
    const component = renderer.create(
      <Provider store={store}>
        <RootContainer />
      </Provider>
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it(`should render its children`, () => {
    const children = (<h1>Test</h1>)
    const component = shallow(
      <Provider store={store}>
        <RootContainer>
          {children}
        </RootContainer>
      </Provider>
    )
    expect(component.contains(children)).toBe(true)
  })
})
