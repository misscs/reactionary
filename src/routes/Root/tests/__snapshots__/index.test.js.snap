// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render 1`] = `
<div>
  <header>
    <div
      className="logo cUhtom"
    >
      <span>
        CC
      </span>
    </div>
    <div
      className="cUhtom"
    >
      Register
    </div>
  </header>
  <main
    id="container"
  />
</div>
`;
